src/components/date-range/date-range.pw.tsx

```diff
@@ -0,0 +1,874 @@
+
+      await expect(startDateElement).toHaveCSS("vertical-align", cssValue);
+
+      const dateRangeComponentLabelElementStartDateParent = dateRangeComponentLabel(
+        page,
+        START_DATE_RANGE_INDEX
+      )
+        .locator("..")
+        .locator("..");
+
+      await expect(dateRangeComponentLabelElementStartDateParent).toHaveCSS(
+        "display",
+        displayValue
+      );
+
+      const endDateElement = page.locator(END_DATE);
+
+      await expect(endDateElement).toHaveCSS("vertical-align", cssValue);
+
+      const dateRangeComponentLabelElementEndDateIndexParent = dateRangeComponentLabel(
+        page,
+        END_DATE_RANGE_INDEX
+      )
+        .locator("..")
+        .locator("..");
+
+      await expect(dateRangeComponentLabelElementEndDateIndexParent).toHaveCSS(
+        "display",
+        displayValue
+      );
+    });
+  });
+
+  test("should check the startDateProps prop", async ({ mount, page }) => {
+    await mount(
+      <DateRangeCustom
+        startDateProps={{
+          disabled: true,
+        }}
+      />
+    );
+
+    const startDateInput = getDataElementByValue(page, "input").nth(
+      START_DATE_RANGE_INDEX
+    );
+
+    await expect(startDateInput).toHaveAttribute("disabled", "");
+  });
+
+  test("should check the endDateProps prop", async ({ mount, page }) => {
+    await mount(
+      <DateRangeCustom
+        endDateProps={{
+          disabled: true,
+        }}
+      />
+    );
+
+    const endDateInput = getDataElementByValue(page, "input").nth(
+      END_DATE_RANGE_INDEX
+    );
+
+    await expect(endDateInput).toHaveAttribute("disabled", "");
+  });
+
+  (["top", "bottom", "left", "right"] as const).forEach((position) => {
+    test(`should check the tooltipPosition is set to ${position}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <DateRangeCustom
+          m={9}
+          tooltipPosition={position}
+          startError={testText}
+        />
+      );
+
+      const iconElement = icon(page);
+      await iconElement.nth(0).hover();
+      const getDataElementByValueElementTooltip = getDataElementByValue(
+        page,
+        "tooltip"
+      );
+      await expect(getDataElementByValueElementTooltip).toHaveAttribute(
+        "data-placement",
+        position
+      );
+
+      await page.hover("body"); // hover on body to close the tooltip
+    });
+  });
+});
+
+test.describe("Event tests for DateRange component", () => {
+  ([0, 1] as const).forEach((inputIndex) => {
+    test(`should call onChange callback when a clear event is triggered on ${inputIndex} DateRange input`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <DateRangeCustom
+          onChange={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      const getDataElementByValueElementInputEqInputIndex = getDataElementByValue(
```

Comment:
**suggestion**: i'm happy with this being declared as a const, but could we not call it `Dateinput`etc or something shorter but still distinct instead of the long const name